pkg/apis/flowcontrol/types.go

```diff
@@ -447,6 +455,43 @@ type LimitedPriorityLevelConfiguration struct {
 	BorrowingLimitPercent *int32
 }
 
+// ExemptPriorityLevelConfiguration describes the configurable aspects
+// of the handling of exempt requests.
+// In the mandatory exempt configuration object the values in the fields
+// here can be modified by authorized users, unlike the rest of the `spec`.
+type ExemptPriorityLevelConfiguration struct {
+	// `nominalConcurrencyShares` (NCS) contributes to the computation of the
+	// NominalConcurrencyLimit (NominalCL) of this level.
+	// This is the number of execution seats nominally reserved for this priority level.
+	// This DOES NOT limit the dispatching from this priority level
+	// but affects the other priority levels through the borrowing mechanism.
+	// The server's concurrency limit (ServerCL) is divided among all the
+	// priority levels in proportion to their NCS values:
+	//
+	// NominalCL(i)  = ceil( ServerCL * NCS(i) / sum_ncs )
+	// sum_ncs = sum[priority level k] NCS(k)
+	//
+	// Bigger numbers mean a larger nominal concurrency limit,
+	// at the expense of every other priority level.
+	// This field has a default value of zero.
+	// +optional
+	NominalConcurrencyShares int32
```

Comment:
So if I'm understanding correctly, this field will be used so that if an exempt priority level uses more than this number of concurrency shares, it will start borrowing from other priority levels.  This is done so that an exempt priority level using greater and greater requests will consume the available concurrency from limited priority levels to keep the overall concurrency at a predictable level.

Even if this priority level runs out of others to borrow from, requests will still be dispatched.

If I've understood correctly, then I need to check the validation and defaulting rules and then I think the API aspect is ok.

Reply:
Primarily you have summarized the feature of borrowing by exempt.  We updated the KEP to describe it, see https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/1040-priority-and-fairness#dispatching .

The default number of nominal concurrency shares for the `exempt` priority level is zero, as documented in the comment on this field.

The reason to allow the `exempt` priority level to have a non-zero number of nominal concurrency shares is so that the effect of its borrowing on other priority levels can be limited by a cluster administrator.  By allocating some concurrency to `exempt`, that means it only borrows when it needs more than that.  The reason that the default is zero is to maintain consistency with the previous shares->absolute mapping.